plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -140,6 +155,51 @@ def associate(source_repo, dest_repo, import_conduit, config, units=None, solver
     return associated_units, failed_units
 
 
+def get_recursive_module_deps(module_units):
+    """
+    Module Metadata specifies dependencies on other modules. This method is useful
+    for taking a set of Modulemd documents and returning the total set of module
+    dependencies for those units.
+
+    :param module_units: iterable of Modulemd
+    :type module_units: iterable
+
+    :return: set of Modulemd
+    :rtype: set
+    """
+    module_deps = set()
+    for unit in module_units:
+        for dependency in unit.dependencies:
+            # we're not a real client, so we don't care about the platform
+            # we need to copy all dependencies for any platform
+            dependency.pop("platform")
+
+            for name, streams in dependency.items():
+                all_module_streams = models.Modulemd.objects.filter(name=name)
+
+                # If there's no stream specified, just copy them all
+                # Better behavior would probably be to copy the default stream, if one exists
+                # TODO?
+                included_streams = [s for s in streams if not s.startswith('-')]
+                if not included_streams:
+                    included_streams = [m.stream for m in all_module_streams]
+
+                # streams preceded by '-' e.g. '-5.21' are not compatible
+                excluded_streams = [
```

Comment:
I think the only time we care about excluded streams is when we don't have the ones to include.
Does it make sense to handle excluded streams only under `if not included_streams` condition?

Reply:
It makes sense I think

Reply:
On second thought that would mean we need 2 different forms of the query.  I think this way is simpler and with loops these small it probably doesn't matter.

Reply:
@dralley, my only worry is the case when/if included and excluded have the same stream. It shouldn't be the case in theory but regardless we need to copy all included ones. I'm ok if we ignore it for now or think about the case as a very rare or impossible one.